aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -65,6 +65,9 @@
     NettyNioAsyncHttpClient(NettySdkHttpClientFactory factory, AttributeMap serviceDefaultsMap) {
         this.serviceDefaults = new ServiceDefaults(serviceDefaultsMap);
         this.trustAllCertificates = factory.trustAllCertificates().orElse(Boolean.FALSE);
+        // 0 thread count will use Netty's default number of threads.
+        this.group = new NioEventLoopGroup(factory.eventLoopConfiguration().numberOfThreads().orElse(0),
+                                           factory.eventLoopConfiguration().threadFactory().orElse(null));
```

Comment:
Should the default thread factory be from the SDK's [ThreadFactoryBuilder](https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/ThreadFactoryBuilder.java) so that we can name the threads? (https://github.com/aws/aws-sdk-java-v2/issues/14)